libopenstorage/operator
# Allow Prometheus customization of replicas,retention,PVC

**What this PR does / why we need it**:
Allow Prometheus customization of 
* replicas
* retention, retentionSize
* PVC, volumeMounts, volumes

**Which issue(s) this PR fixes** (optional)
* [Ticket](https://portworx.atlassian.net/browse/PWX-24897)
* Previous [PR](https://github.com/libopenstorage/operator/pull/831) to customize Resources and SecurityContext

**Special notes for your reviewer**:

Test:

1. In StorageCluster spec, specify replicas, retention, retentionSize, storage, volumeMounts, volumes

<img width="372" alt="Screen Shot 2023-02-22 at 7 16 49 PM" src="https://user-images.githubusercontent.com/122411280/220814228-09a4ddc9-a66c-40eb-ac08-fd29cbc565a5.png">

2. Wait for a few seconds, delete px-operator pod
3. Verify prometheus spec has been updated, and 2 prometheus pods running
<img width="307" alt="Screen Shot 2023-02-22 at 7 21 50 PM" src="https://user-images.githubusercontent.com/122411280/220814585-7ddff6ad-e14b-444e-a65e-02ad7e59bdfd.png">

4. Describe prometheus pod
<img width="345" alt="Screen Shot 2023-02-22 at 7 27 14 PM" src="https://user-images.githubusercontent.com/122411280/220815791-e8b5cff5-2172-4a68-9a78-9755595eaea9.png">
<img width="572" alt="Screen Shot 2023-02-22 at 7 32 53 PM" src="https://user-images.githubusercontent.com/122411280/220815798-f185021a-d37a-48bb-aa03-1ab19e8b8826.png">
<img width="1180" alt="Screen Shot 2023-02-22 at 7 33 03 PM" src="https://user-images.githubusercontent.com/122411280/220815811-4c8722ab-3238-4306-8e9d-c10b19ddbf4d.png">
